<launch>
  <!-- smartcar -->
	<arg name="config_param_file" default="$(find smartcar_bringup)/param/controller_param.yaml"/>

	<!-- node的三个属性分别为程序包名字pkg，可执行文件名type和节点名字的名字name--> 
	<node pkg="smartcar_bringup" type="smartcar_node2" name="smartcar_controller2" output="screen">

		<rosparam file="$(arg config_param_file)" command="load" />
	</node>

    	<!-- Load the URDF/Xacro model of our robot -->  
	<param name="robot_description" command="$(find xacro)/xacro.py '$(find smartcar_description)/urdf/robots/smartcarV2.xacro'" />  

	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >  
		<param name="use_gui" value="False"/>
	</node>  

	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">  
		<param name="publish_frequency" type="double" value="20.0" />  
	</node>  

</launch>
